src/main/scala/linting/rule/LintConflictingModuleNames.scala

```diff
@@ -0,0 +1,314 @@
+  * module name conflicts will cause a [[LintViolation]]. If one of the modules
+  * is targeted by [[NamingStrategyAnnotation]] then the transform will attempt
+  * to rename modules according to that naming strategy. If the conflicting
+  * modules cannot be disambiguated by the naming strategy then a
+  * [[LintViolation]] is emitted.
+  */
+final class LintConflictingModuleNames extends LintRule {
+  val recommendedFix: String = "override desiredName based on module parameters"
+
+  val lintName: String = "conflicting-module-names"
+
+  override def optionalPrerequisites: Seq[Dependency[Transform]] =
+    Dependency[RenameDesiredNames] +: super.optionalPrerequisites
+
+  override def execute(state: CircuitState): CircuitState = {
+    val violations = new Violations()
+
+    val modMap = state.circuit.modules.collect {
+      case m: Module => m.name -> m
+    }.toMap
+
+    val desiredNameAnnos = state.annotations.collect {
+      case a: DesiredNameAnnotation if a.target.circuit == state.circuit.main => a
+    }
+
+    val moduleToDesiredName: mutable.Map[String, mutable.Set[String]] = mutable.Map()
+
+    val nameMap = desiredNameAnnos.groupBy(_.desiredName).mapValues { annos =>
+      annos.map(a => Target.referringModule(a.target).module).distinct.map { referringModule =>
+        require(modMap.contains(referringModule), "ModuleNameAnnotations may not refer to blackboxes")
+        val desiredNames = moduleToDesiredName.getOrElseUpdate(referringModule, mutable.Set())
+        desiredNames += annos.head.desiredName
+        modMap(referringModule)
+      }
+    }
+
+    val conflictingDesiredNames = moduleToDesiredName.collect {
+      case kv@ (moduleName, desiredName) if desiredName.size > 1 => kv
+    }
+
+    require(conflictingDesiredNames.size == 0, {
+      val explanation = conflictingDesiredNames.map {
+        case (modName, desiredNames) => s"  ${modName}: ${desiredNames.mkString(", ")}"
+      }.mkString("\n")
+      s"Modules may not have more than one desiredName:\n${explanation}"
+    })
+
+    nameMap.foreach {
+      case (desiredName, modules) if modules.size > 1 =>
+        val msg = s"Module conflicts for desired name $desiredName: ${modules.map(_.name).mkString(", ")}"
+        val info = MultiInfo(modules.map(_.info))
+        val mods = violations.getOrElse((info, msg), Set.empty)
+        violations((info, msg)) = mods ++ modules.map(_.name)
+      case _ =>
+    }
+
+    val whitelist = collectWhitelist(state.annotations)
+    val errorList = violations.collect {
+      case ((info, message), mods) if !isWhitelisted(info, whitelist) => Violation(this, info, message, mods)
+    }.toSeq.sortBy { _.toString }
+    val newAnnos = errorList ++ state.annotations
+
+    state.copy(annotations = newAnnos)
+  }
+}
+
+/** Renames modules based on their overridden desired names
+  *
+  * Desired module name overrides are specified by
+  * [[OverrideDesiredNameAnnotation]]. If one of the modules is targeted by a
```

Comment:
Is this commentary still accurate? I thought this was all done in the above

Reply:
I guess what I mean, shouldn't the renaming happen only here, but not in the linting step?

Reply:
this is what the current behavior is. `LintConflictingModuleNames` only checks `DesiredNameAnnotations` and `RenameDesiredNames` does the renaming according to `OverrideDesiredNameAnnotation`

Reply:
yeah i was confused because it was in the other palce as well, but it looks like you removed it from the earlier scaladoc, so I will resolve this comment now